src/parameterizations/vertical/MOM_geothermal.F90

```diff
@@ -175,6 +192,12 @@ subroutine geothermal(h, tv, dt, ea, eb, G, GV, CS, halo)
     do k=nz,1,-1
       do i=isj,iej ; if (do_i(i)) then
 
+        ! Save temperature and thickness before any changes are made (for diagnostic)
+        if (CS%id_internal_heat_tend_3d > 0) then
+          T_old = tv%T(i,j,k)
```

Comment:
Does `T_old` need indices here?

Reply:
Currently just have T_old as a scalar to avoid allocating another 3D array. I will do it in 3D when I code up a temperature diagnostic.

Reply:
Right, I see now.  Sorry about that!